File: experiments/control-plane-staging/versions/control-plane-opt-recovered-decompiled-20260222-192441/snapshot/control-plane/src/test/java/it/unimib/datai/nanofaas/controlplane/execution/ExecutionRecordLegacyAccessorsTest.java
package it.unimib.datai.nanofaas.controlplane.execution;

import it.unimib.datai.nanofaas.common.model.ErrorInfo;
import it.unimib.datai.nanofaas.controlplane.scheduler.InvocationTask;
import org.junit.jupiter.api.Test;

import java.time.Instant;

import static org.assertj.core.api.Assertions.assertThat;

class ExecutionRecordLegacyAccessorsTest {

    @Test
    void legacyMutators_updateFieldsConsistently() {
        InvocationTask originalTask = new InvocationTask("exec-1", "fn", null, null, null, null, null, 1);
        ExecutionRecord record = new ExecutionRecord("exec-1", originalTask);

        InvocationTask updatedTask = new InvocationTask("exec-1", "fn", null, null, null, null, null, 2);
        Instant startedAt = Instant.now().minusSeconds(5);
        Instant finishedAt = Instant.now();
        ErrorInfo error = new ErrorInfo("E", "boom");

        record.updateTask(updatedTask);
        record.state(ExecutionState.RUNNING);
        record.startedAt(startedAt);
        record.output("payload");
        record.lastError(error);
        record.finishedAt(finishedAt);

        assertThat(record.task()).isEqualTo(updatedTask);
        assertThat(record.state()).isEqualTo(ExecutionState.RUNNING);
        assertThat(record.startedAt()).isEqualTo(startedAt);
        assertThat(record.finishedAt()).isEqualTo(finishedAt);
        assertThat(record.output()).isEqualTo("payload");
        assertThat(record.lastError()).isEqualTo(error);
    }

    @Test
    void markSuccess_clearsError_andMarkError_clearsOutput() {
        ExecutionRecord record = new ExecutionRecord(
                "exec-2",
                new InvocationTask("exec-2", "fn", null, null, null, null, null, 1)
        );

        record.markRunning();
        record.markError(new ErrorInfo("ERR", "first"));
        assertThat(record.lastError()).isNotNull();
        assertThat(record.output()).isNull();

        record.markSuccess("ok");
        assertThat(record.output()).isEqualTo("ok");
        assertThat(record.lastError()).isNull();

        record.markError(new ErrorInfo("ERR2", "second"));
        assertThat(record.output()).isNull();
        assertThat(record.lastError().code()).isEqualTo("ERR2");
    }
}

File: experiments/control-plane-staging/versions/control-plane-rust-m3-20260222-200159/snapshot/control-plane-rust/tests/execution_record_legacy_accessors_test.rs
#![allow(non_snake_case)]

use control_plane_rust::execution::{ErrorInfo, ExecutionRecord, ExecutionState, InvocationTask};

#[test]
fn legacyMutators_updateFieldsConsistently() {
    let original_task = InvocationTask::new("exec-1", "fn", 1);
    let mut record = ExecutionRecord::new_with_task("exec-1", original_task.clone());

    let updated_task = InvocationTask::new("exec-1", "fn", 2);

    record.update_task(updated_task.clone());
    record.set_state(ExecutionState::Running);
    record.set_started_at_millis(Some(10));
    record.set_output(Some(serde_json::json!("payload")));
    record.set_last_error(Some(ErrorInfo::new("E", "boom")));
    record.set_finished_at_millis(Some(20));

    assert_eq!(record.task(), &updated_task);
    assert_eq!(record.state(), ExecutionState::Running);
    assert_eq!(record.started_at_millis(), Some(10));
    assert_eq!(record.finished_at_millis(), Some(20));
    assert_eq!(record.output(), Some(serde_json::json!("payload")));
    assert_eq!(record.last_error(), Some(ErrorInfo::new("E", "boom")));
}

#[test]
fn markSuccess_clearsError_andMarkError_clearsOutput() {
    // Test that mark_success_at clears last_error (valid path: Queued -> Running -> Success)
    let mut record =
        ExecutionRecord::new_with_task("exec-2", InvocationTask::new("exec-2", "fn", 1));
    record.mark_running_at(10);
    record.set_last_error(Some(ErrorInfo::new("ERR", "first")));
    assert!(record.last_error().is_some());
    record.mark_success_at(serde_json::json!("ok"), 30);
    assert_eq!(record.output(), Some(serde_json::json!("ok")));
    assert!(record.last_error().is_none());

    // Test that mark_error_at clears output (valid path: Queued -> Running -> Error)
    let mut record2 =
        ExecutionRecord::new_with_task("exec-2b", InvocationTask::new("exec-2b", "fn", 1));
    record2.mark_running_at(10);
    record2.set_output(Some(serde_json::json!("previous")));
    assert!(record2.output().is_some());
    record2.mark_error_at(ErrorInfo::new("ERR2", "second"), 40);
    assert!(record2.output().is_none());
    assert_eq!(
        record2.last_error().map(|e| e.code),
        Some("ERR2".to_string())
    );
}
